src/components/ThemeSwitcher.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { useEffect, useState } from 'react';
+
+import { HiOutlineMoon, HiOutlineSun } from 'react-icons/hi/index';
+
+const ThemeSwitcher = () => {
+  const getInitialState = () => {
+    if (typeof window !== 'undefined') {
+      return localStorage.getItem('theme') as 'light' | 'dark' | undefined;
+    }
+    return 'light';
+  };
+  const [theme, setTheme] = useState<'light' | 'dark' | undefined>(
+    getInitialState
```

Comment:
This is causing hydration issues(notice errors in browser console) as we pre-render the pages and don't know what the theme could be.
Can we move this to a `useEffect` like we had before?

```suggestion
    const [theme, setTheme] = useState<'light' | 'dark'>('light');
    
    useEffect(() => {
    const localTheme = localStorage.getItem('theme') as 'light' | 'dark' | null;
    setTheme(localTheme ?? 'light');
  }, []);
```